### ISSUE TITLE
GdipCombineRegionPath assertion failed

### ISSUE THREAD

**Issue by joshswimlane** [vexx32/PSWordCloud#14]
Hello, when trying to run PSWordCloud on my Mac I am running into this issue:

* installed `mono-libgdiplus`
* on macOS Mojave (10.14.2)
* pwsh version (6.1.1)

Running the following:
```
$get_domain = Get-Content -Path /Users/josh.rickard/get_domain.json
New-WordCloud -Path /Users/josh.rickard/Desktop/get_domain.png -Input $get_domain
```

I receive the following error, which dumps me out of my `pwsh` session and provides the following information:
```
** (process:19320): WARNING **: 14:34:29.106: Path conversion requested 0 bytes (8388608 x 8388608). Maximum size is 8388608 bytes.       
** ERROR:region.c:1155:GdipCombineRegionPath: assertion failed: (region->bitmap)   Abort trap: 6 
```

Any ideas?

**Comment by vexx32** [comment]
That's an interesting error. I don't have a Mac myself, so it's a bit dicey to troubleshoot.

However... that number seems _awfully_ familiar, that 8388608 number. I think that's the upper bound on canvas or region size in System.Drawing....

Makes me wonder if perhaps the Mac implementation of System.Drawing doesn't support "infinite" regions, which Windows handles just fine.  Linux has a similar issue, but consistently and always crashes pwsh. Some Mac versions / setups seem to be OK handling it, although far from perfect (cc @TylerLeonhardt , who's been helping me look at the xplat issues a bit and actually has a Mac to test with.)

Can you provide a copy of the input data you gave it?

---

I may or may not end up fixing this issue properly; I'm planning on simply converting this right on over to a compiled cmdlet based on `SkiaSharp`, which is supposed to work properly across all of Windows/Mac/Linux... I hope!

It'll also enable me to draw things with vector graphics, which should be fun! 😄

**Comment by TylerLeonhardt** [comment]
I remember it working a couple times when I specified the `-ImageSize` @joshswimlane maybe try setting that?

**Comment by joshswimlane** [comment]
@TylerLeonhardt  I tried it with the `-ImageSize 1080p` as well and same error.  I have tried it with just a text file called `test.txt` with the following included:

```
powershell python use case wonderful logic simple touch case python underscore blah blah
```

Still, receive the same error.

**Comment by TylerLeonhardt** [comment]
This shouldn't be an issue after the switch to Skia, right @vexx32?

**Comment by vexx32** [comment]
Yeah this is specific to System.Drawing.

Once I get everything ironed out in the c# or f# versions I'll close this.

**Comment by vexx32** [comment]
Resolved in v2.0 with the move to SkiaSharp. 😄
